pkg/operator/rules.go

```diff
@@ -86,16 +86,20 @@ func (prs *PrometheusRuleSelector) generateRulesConfiguration(promRule *monitori
 			promRuleSpec.Groups[i].Limit = nil
 			level.Warn(logger).Log("msg", fmt.Sprintf("`limit` is supported only from %s version %q", component, minVersionLimits))
 		}
+
 		if prs.ruleFormat == PrometheusFormat {
 			// Unset partialResponseStrategy field.
 			promRuleSpec.Groups[i].PartialResponseStrategy = ""
 		}
 	}
 
+	promRuleSpec = prs.validateRuleFieldsWithVersion(promRuleSpec)
```

Comment:
I'd move L84-L94 into `validateRuleFieldsWithVersion()` too

Reply:
aah correct I thought same, I will move it